### DISCUSSION TITLE
How do I define a job with a multi-partitioned asset and its upstream dependencies

### DISCUSSION THREAD

**Discussion by tacastillo** [dagster-io/dagster#15247] [Q&A]
Reposting on the behalf of a user:

I have a multi-partitioned asset that has depend on two partitioned assets with different PartitionsDefinitions. How do I define a job for it?

Passing any of the partition definitions results in similar exceptions:
```
dagster._core.errors.DagsterInvalidDefinitionError: Multiple partitioned assets exist in assets job 'test_job'. Selected assets must have the same partitions definitions, but the selected assets have different partitions definitions:
```

Below is a repro:

```python
from dagster import (
    DailyPartitionsDefinition,
    MultiPartitionsDefinition,
    StaticPartitionsDefinition,
    asset, define_asset_job,
    Definitions, AutoMaterializePolicy, AssetSelection, in_process_executor,
)

datep = DailyPartitionsDefinition(start_date="2023-06-23")
colorp = StaticPartitionsDefinition(["red", "yellow", "blue", "green", "black", "white", "orange"])
shapesp = StaticPartitionsDefinition(["square", "round"])
multip = MultiPartitionsDefinition({"up_a": datep, "up_b": colorp})

@asset(partitions_def=datep)
def up_a():
    return [1, 2, 3]

@asset(partitions_def=colorp)
def up_b():
    return ['a', 'b']

@asset(
    partitions_def=multip,
    auto_materialize_policy=AutoMaterializePolicy.eager(),
)
def multi_partitions_asset(context, up_a, up_b):
    return up_a + up_b

myjob = define_asset_job('test_job',
    selection=AssetSelection.assets(multi_partitions_asset, up_a, up_b),
    partitions_def=multip,
    executor_def=in_process_executor,
)

defs = Definitions(
    jobs=[myjob],
    assets=[up_a, up_b, multi_partitions_asset],
)
```

**Comment by sryza** [comment]
It's not currently possible to have a job that targets assets with different `PartitionsDefinition`s.  Here's an issue where we're tracking this: https://github.com/dagster-io/dagster/issues/13257.

**Reply by someaveragepunter** [reply]
thanks @sryza 
adding @OwenKephart  @slopp  who understand my usecase

context:
my actual goal is just to be able to run a partitioned dag in process using in_process_executor
more specifically I'd like my "job" to eagerly materialize such that if any of the input edges updates via a sensor, the entire affected downstream dag will rerun to update but all within the same python process (to avoid overhead)

appreciate you have some edge cases to deal with in https://github.com/dagster-io/dagster/issues/13257 when it comes to datetime partitions but for static ones I'd have thought there's no ambiguity and should be supported?

![image](https://github.com/dagster-io/dagster/assets/133558867/50e736b8-6b08-4654-a42b-0e63e9bcb9bc)
